src/main/java/bio/terra/service/filedata/azure/tables/TableFileDao.java

```diff
@@ -0,0 +1,146 @@
+package bio.terra.service.filedata.azure.tables;
+
+import bio.terra.service.filedata.exception.FileSystemCorruptException;
+import bio.terra.service.filedata.google.firestore.*;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.data.tables.TableClient;
+import com.azure.data.tables.TableServiceClient;
+import com.azure.data.tables.models.ListEntitiesOptions;
+import com.azure.data.tables.models.TableEntity;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+
+import com.google.api.core.SettableApiFuture;
+import com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+/**
+ * TableFileDao provides CRUD operations on the "files" storage table in Azure. Objects in the
+ * file table are referred to by the dataset (owner of the files) and by any snapshots that
+ * reference the files. File naming is handled by the directory DAO. This DAO just handles the basic
+ * operations for managing the collection. It does not have logic for protecting against deleting
+ * files that have dependencies or creating files with duplicate paths.
+ *
+ */
+@Component
+class TableFileDao {
+    private final Logger logger = LoggerFactory.getLogger(TableFileDao.class);
+    private final FireStoreUtils fireStoreUtils;
```

Comment:
I think it may be time to rename `FireStore...` prefixed classes if we're gonna use them with Azure too. Maybe `DataRepoFileSystemUtils`, `DataRepoFile`, `DataRepoDirectory`, etc...

Reply:
Yeah that's a good point

Reply:
This no longer uses the firestore utils btw!